nimbus-core/src/main/java/com/antheminc/oss/nimbus/domain/config/builder/DefaultAnnotationConfigHandler.java

```diff
@@ -134,26 +134,72 @@ protected AnnotationConfig handleSingleInternal(AnnotatedElement aElem, Class<?
 				
 				// check that the declared annotation has passed in meta annotation type
 				
-				boolean hasRepeatable = AnnotatedElementUtils.hasAnnotation(repeatableAnnotationDeclared, repeatableMetaAnnotationType);
+				boolean hasRepeatable = AnnotatedElementUtils.hasAnnotation(repeatableAnnotationDeclared, Event.class);
```

Comment:
I think its time to start separating out logic that is specific to Event from this class which is generic with a purpose to handle the pattern of repeatable annotation with handlers. We may find a scenario to use the same for something else other than Event.

Could we consider using a strategy pattern to handle just Event?